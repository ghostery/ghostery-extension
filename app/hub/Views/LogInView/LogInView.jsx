/**
 * LogIn View
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';
import { NavLink } from 'react-router-dom';

/**
 * A Functional React component for rendering the Login View
 * @return {JSX} JSX for rendering the Login View of the Hub app
 * @memberof HubComponents
 */
const LogInView = (props) => {
	const {
		email,
		password,
		emailError,
		passwordError,
		handleSubmit,
		handleInputChange,
	} = props;

	const emailInputClassNames = ClassNames('LogInView__inputBox', {
		error: emailError,
	});
	const passwordInputClassNames = ClassNames('LogInView__inputBox', {
		error: passwordError,
	});

	return (
		<div className="LogInView">
			<div className="LogInView--addPaddingTop row align-center">
				<div className="columns">
					<div className="LogInView__header flex-container align-center-middle">
						<img className="LogInView__headerImage" src="/app/images/hub/account/ghosty-account.svg" />
						<div className="LogInView__headerTitle text-center">
							<h3>
								{t('hub_login_header_title')}
							</h3>
						</div>
					</div>
				</div>
			</div>
			<div className="LogInView--addPaddingTop row align-center">
				<div className="columns small-12 medium-10 large-6">
					<form onSubmit={handleSubmit}>
						<label htmlFor="login-email" className="LogInView__inputLabel">
							{t('email_colon')}
						</label>
						<input
							id="login-email"
							className={emailInputClassNames}
							name="email"
							type="text"
							value={email}
							onChange={handleInputChange}
							pattern=".{1,}"
							autoComplete="off"
						/>
						{emailError && (
							<div className="LogInView__inputError">
								{t('please_enter_a_valid_email')}
							</div>
						)}
						<label htmlFor="login-password" className="LogInView__inputLabel">
							{t('password_colon')}
						</label>
						<input
							id="login-password"
							className={passwordInputClassNames}
							name="password"
							type="password"
							value={password}
							onChange={handleInputChange}
							pattern=".{1,}"
						/>
						{passwordError && (
							<div className="LogInView__inputError">
								{t('hub_login_label_password_invalid')}
							</div>
						)}
						<div className="flex-container align-center">
							<span className="LogInView__link text-center">
								<NavLink to="/forgot-password">
									{ t('forgot_password') }
								</NavLink>
							</span>
						</div>
						<div className="flex-container align-center">
							<span className="LogInView__link text-center">
								{ t('hub_login_link_dont_have_account') }
								&nbsp;
								<NavLink to="/create-account">
									{ t('hub_login_link_create_account') }
								</NavLink>
							</span>
						</div>
						<div className="LogInView--addPaddingTop flex-container align-center">
							<button type="submit" className="LogInView__button button success">
								{ t('sign_in') }
							</button>
						</div>
					</form>
				</div>
			</div>
		</div>
	);
};

// PropTypes ensure we pass required props of the correct type
LogInView.propTypes = {
	email: PropTypes.string.isRequired,
	password: PropTypes.string.isRequired,
	emailError: PropTypes.bool.isRequired,
	passwordError: PropTypes.bool.isRequired,
	handleSubmit: PropTypes.func.isRequired,
	handleInputChange: PropTypes.func.isRequired,
};

export default LogInView;
