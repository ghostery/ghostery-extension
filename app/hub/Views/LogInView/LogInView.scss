/**
 * Log In View Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

// Log In View
.LogInView {
	margin-top: 40px;
	padding-bottom: 40px;
}
.LogInView--addPaddingTop {
	padding-top: 40px;
}
.LogInView__headerImage {
	max-width: 180px;
}
.LogInView__headerTitle h3 {
	font-size: 24px;
	line-height: 1.5;
	color: #4a4a4a;
	margin: 20px 25px 0;
}
.LogInView__inputLabel {
	font-size: 14px;
	font-weight: 500;
	line-height: 40px;
	color: #4a4a4a;
}
.LogInView__inputBox {
	font-size: 14;
	line-height: 24px;
	color: #4a4a4a;
	margin-bottom: 35px;

	// Foundation Overrides
	border-radius: 0;
	box-shadow: none;
	border: 1px solid #c8c7c2;
}
.LogInView__inputBox.error {
	margin-bottom: 8px;
	border-color: #e74055;
}
.LogInView__inputBox:focus {
	// Foundation Overrides
	box-shadow: none;
	border-color: #4a4a4a;
}
.LogInView__inputError {
	font-size: 12;
	line-height: 14px;
	color: #e74055;
	margin-bottom: 13px;
}
.LogInView__link {
	font-size: 14px;
	line-height: 30px;
}
.LogInView__button {
	min-width: 180px;
}
@media only screen and (max-width: 740px)  {
	.LogInView__header {
		flex-direction: column;
	}
}
