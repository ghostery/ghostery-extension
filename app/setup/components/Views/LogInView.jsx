/**
 * Log In View Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import { log } from '../../../../src/utils/common';
import { utils } from '../../utils';

/**
 * @class Implement the #log-in part of the Setup flow.
 * Allows users to log in or to create an account.
 * @extends Component
 * @memberof SetupViews
 */
class LogInView extends Component {
	constructor(props) {
		super(props);
		this.state = {
			status: '', // 'new', 'login' or 'loggedin'
			email: '',
			confirmEmail: '',
			password: '',
			firstName: '',
			lastName: '',
		};
	}

	/**
	* Lifecycle event
	*/
	componentWillMount() {
		this.props.actions.reset();
		this.props.actions.getUser()
			.then((res) => {
				const { errors, user } = res;
				if (errors) {
					this.setState({ status: 'new' });
					this.props.actions.updateNavigationNextButtons([
						{
							title: t('setup_button_skip'),
							action: 'next',
						}, {
							title: t('setup_button_create_account'),
							action: 'createAccount',
						},
					]);
				} else {
					this.setState({
						status: 'loggedin',
						email: user.email,
					});
					this.props.actions.updateNavigationNextButtons([
						{
							title: t('setup_button_next'),
							action: 'next',
						},
					]);
				}
			});
		this.props.actions.updateTopContentData({
			image: '/app/images/setup/circles/cloud.svg',
			title: t('setup_login_view_title'),
		});
	}

	/**
	* Lifecycle event
	*/
	componentDidMount() {
		this.props.actions.setupStep({ key: 'setup_step', value: 4 });
	}

	/**
	* Lifecycle event
	*/
	componentWillReceiveProps(nextProps) {
		if (nextProps.triggerSignIn) {
			this._signIn();
		} else if (nextProps.triggerCreateAccount) {
			this._createAccount();
		}

		if (!nextProps.loading) {
			this.props.actions.hideLoading();
		} else {
			this.props.actions.showLoading();
		}

		if ((this.state.status === 'new' || this.state.status === 'login') && nextProps.success) {
			this.props.actions.navigationNext();
		}
		// TODO
		// if (nextProps.message !== this.props.message) {
		// 	window.alert(nextProps.message);
		// }
	}

	/**
	 * Handles the onSubmit property on the Sign In form
	 * @param  {Object} event
	 */
	_signIn = (e) => {
		if (e) { e.preventDefault(); }
		const { email, password } = this.state;
		if (!this._validateSignIn()) {
			return;
		}

		this.props.actions.login(email, password)
			.then((success) => {
				if (success) {
					Promise.all([
						this.props.actions.getUser(),
						this.props.actions.getUserSettings(),
					]);
				}
			})
			.catch(err => log(err));

		this.props.actions.showLoading();
	}


	/**
	 * Handles the onSubmit property on the Create Account form
	 * @param  {Object} event
	 */
	_createAccount = (e) => {
		if (e) { e.preventDefault(); }
		if (!this._validateCreateAccount()) {
			return;
		}

		const {
			email, confirmEmail, firstName, lastName, password
		} = this.state;
		this.props.actions.register(email, confirmEmail, firstName, lastName, password)
			.then((success) => {
				if (success) {
					this.props.actions.getUser();
				}
			});
		this.props.actions.showLoading();
	}

	/**
	 * Handles the onClick property to show the Create Account form
	 * @param  {Object} event
	 */
	_toggleSignIn = (event) => {
		event.preventDefault();
		const { status } = this.state;
		let title;
		let action;
		let nextStatus;

		if (status === 'new') {
			nextStatus = 'login';
		} else if (status === 'login') {
			nextStatus = 'new';
		}

		if (nextStatus === 'loggedin') {
			this.props.actions.updateNavigationNextButtons([
				{
					title: t('setup_button_next'),
					action: 'next',
				},
			]);
		} else {
			switch (nextStatus) {
				case 'new':
					title = t('setup_button_create_account');
					action = 'createAccount';
					break;
				case 'login':
					title = t('setup_button_sign_in');
					action = 'signIn';
					break;
				default:
					break;
			}
			this.props.actions.updateNavigationNextButtons([
				{
					title: t('setup_button_skip'),
					action: 'next',
				}, {
					title,
					action,
				},
			]);
		}

		if (typeof nextStatus !== 'undefined') {
			this.setState({
				status: nextStatus,
			});
		}
	}

	/**
	 * Validates the Sign In form
	 * @return {boolean} whether the form is valid
	 */
	_validateSignIn() {
		this.emailInput.focus();
		this.emailInput.blur();
		this.passwordInput.focus();
		this.passwordInput.blur();
		return (
			this._isValid({
				target: {
					name: 'email',
					value: this.state.email,
				},
			}) &&
			this._isValid({
				target: {
					name: 'password',
					value: this.state.password,
				},
			}) || false);
	}

	/**
	* Validates the Create Account form
	* @return {boolean} whether the form is valid
	*/
	_validateCreateAccount() {
		this.emailInput.focus();
		this.emailInput.blur();
		this.confirmEmailInput.focus();
		this.confirmEmailInput.blur();
		this.passwordInput.focus();
		this.passwordInput.blur();
		return (
			this._isValid({
				target: {
					name: 'email',
					value: this.state.email,
				},
			}) &&
			this._isValid({
				target: {
					name: 'confirmEmail',
					value: this.state.confirmEmail,
				},
			}) &&
			this._isValid({
				target: {
					name: 'password',
					value: this.state.password,
				},
			}));
	}

	/**
	 * Used for validating forms
	 * @param  {Object} event
	 * @return {boolean} whether the form is valid
	 */
	_isValid = (e) => {
		const { name, value } = e.target;
		const { email, password } = this.state;
		switch (name) {
			case 'email':
				return utils.validateEmail(value);
			case 'confirmEmail':
				return email === value;
			case 'password':
				return utils.validatePassword(password);
			default:
				return true;
		}
	}

	/**
	 * handles the onChange property
	 * @param  {Object} event
	 */
	_onChange = (e) => {
		const { name, value } = e.target;
		this.setState({ [name]: value });
	}

	/**
	 * handles the onBlur property
	 * @param  {Object} event
	 */
	_onBlur = (e) => {
		if (this._isValid(e)) {
			e.target.classList.remove('invalid');
		} else {
			e.target.classList.add('invalid');
		}
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the #log-in step of the setup flow
	 */
	render() {
		return (
			<div id="log-in-view">
				{this.state.status === 'new' &&
					<form onSubmit={this._createAccount} noValidate>
						<div className="row align-center new-account">
							<div className="columns small-10 large-8">
								<div className="row small-up-1 medium-up-2 align-center">
									<div className="columns">
										<label>{ t('setup_login_view_email') }</label>
										<input
											name="email"
											type="email"
											value={this.state.email}
											onChange={this._onChange}
											onBlur={this._onBlur}
											required
											ref={(ref) => { this.emailInput = ref; }}
										/>
										<span className="form-error-container">
											<span className="form-error">{ t('setup_login_view_email_error') }</span>
										</span>
									</div>
									<div className="columns">
										<label>{ t('setup_login_view_confirm_email') }</label>
										<input
											name="confirmEmail"
											type="email"
											value={this.state.confirmEmail}
											onChange={this._onChange}
											onBlur={this._onBlur}
											required
											ref={(ref) => { this.confirmEmailInput = ref; }}
										/>
										<span className="form-error-container">
											<span className="form-error">{ t('setup_login_view_confirm_email_error') }</span>
										</span>
									</div>
									<div className="columns">
										<label>{ t('setup_login_view_first_name') }</label>
										<input
											name="firstName"
											type="text"
											value={this.state.firstName}
											onChange={this._onChange}
											onBlur={this._onBlur}
										/>
									</div>
									<div className="columns">
										<label>{ t('setup_login_view_last_name') }</label>
										<input
											name="lastName"
											type="text"
											value={this.state.lastName}
											onChange={this._onChange}
											onBlur={this._onBlur}
										/>
									</div>
									<div className="columns">
										<label>{ t('setup_login_view_password') }</label>
										<input
											name="password"
											type="password"
											value={this.state.password}
											onChange={this._onChange}
											onBlur={this._onBlur}
											required
											ref={(ref) => { this.passwordInput = ref; }}
										/>
										<span className="form-error-container">
											<span className="form-error">{ t('setup_login_view_password_error') }</span>
										</span>
									</div>
									<div className="columns" />
									<div className="columns">
										<a onClick={this._toggleSignIn}>{ t('setup_login_view_log_in') }</a>
									</div>
									<div className="columns" />
								</div>
							</div>
						</div>
						<input type="submit" className="hide" />
					</form>
				}
				{this.state.status === 'login' &&
					<form onSubmit={this._signIn} noValidate>
						<div className="row align-center new-account">
							<div className="columns small-10 medium-8 large-6">
								<div className="row small-up-1 align-center">
									<div className="columns">
										<label>{ t('setup_login_view_email') }</label>
										<input
											name="email"
											type="email"
											value={this.state.email}
											onChange={this._onChange}
											onBlur={this._onBlur}
											required
											ref={(ref) => { this.emailInput = ref; }}
										/>
										<span className="form-error-container">
											<span className="form-error">{ t('setup_login_view_email_error') }</span>
										</span>
									</div>
									<div className="columns">
										<label>{ t('setup_login_view_password') }</label>
										<input
											name="password"
											type="password"
											value={this.state.password}
											onChange={this._onChange}
											onBlur={this._onBlur}
											required
											ref={(ref) => { this.passwordInput = ref; }}
										/>
										<span className="form-error-container">
											<span className="form-error">{ t('setup_login_view_password_error') }</span>
										</span>
									</div>
									<div className="columns">
										<a onClick={this._toggleSignIn}>{ t('setup_login_view_create_account') }</a>
									</div>
								</div>
								<input type="submit" className="hide" />
							</div>
						</div>
					</form>
				}
				{this.state.status === 'loggedin' &&
					<div className="row">
						<div className="columns text-center logged-in">
							<h3>
								{t('setup_login_view_signed_in_as')} {this.state.email}
							</h3>
						</div>
					</div>
				}
			</div>
		);
	}
}

export default LogInView;
