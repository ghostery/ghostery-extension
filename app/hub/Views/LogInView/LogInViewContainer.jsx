/**
 * LogIn View Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { validateEmail } from '../../../panel/utils/utils';
import LogInView from './LogInView';
import SignedInView from '../SignedInView';
/**
 * @class Implement the Log In View for the Ghostery Hub
 * @extends Component
 * @memberof HubContainers
 */
class LogInViewContainer extends Component {
	constructor(props) {
		super(props);
		this.state = {
			email: '',
			password: '',
			emailError: false,
			passwordError: false,
			validateInput: false,
		};

		this.props.actions.setToast({
			toastMessage: '',
			toastClass: '',
		});
	}

	/**
	 * Update input values by updating state.
	 * @param  {Object} event the 'change' event
	 */
	_handleInputChange = (event) => {
		const { name, value } = event.target;
		this.setState({ [name]: value });

		if (!this.state.validateInput) {
			return;
		}

		switch (name) {
			case 'email': {
				const emailIsValid = value && validateEmail(value);
				this.setState({
					emailError: !emailIsValid,
				});
				break;
			}
			case 'password': {
				this.setState({
					passwordError: !value,
				});
				break;
			}
			default: break;
		}
	}

	/**
	 * Handle logging in, but validate the data first.
	 * @param  {Object} event the 'submit' event
	 */
	_handleLoginAttempt = (event) => {
		event.preventDefault();
		const { email, password } = this.state;
		const emailIsValid = email && validateEmail(email);

		this.setState({
			emailError: !emailIsValid,
			passwordError: !password,
			validateInput: true,
		});

		if (!emailIsValid || !password) {
			return;
		}

		this.props.actions.setToast({
			toastMessage: '',
			toastClass: ''
		});
		this.props.actions.login(email, password).then((success) => {
			if (success) {
				const { origin, pathname, hash } = window.location;
				window.history.pushState({}, '', `${origin}${pathname}${hash}`);

				this.props.actions.getUser();
				this.props.actions.getUserSettings()
					.then((settings) => {
						const { current_theme } = settings;
						return this.props.actions.getTheme(current_theme);
					});
				this.props.actions.setToast({
					toastMessage: t('hub_login_toast_success'),
					toastClass: 'success'
				});
				this.props.history.push('/');
			} else {
				this.props.actions.setToast({
					toastMessage: t('hub_login_toast_error'),
					toastClass: 'alert'
				});
			}
		});
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Log In View of the Hub app
	 */
	render() {
		const { loggedIn, user } = this.props;
		const {
			email,
			password,
			emailError,
			passwordError,
		} = this.state;
		const logInChildProps = {
			email,
			password,
			emailError,
			passwordError,
			handleInputChange: this._handleInputChange,
			handleSubmit: this._handleLoginAttempt,
		};
		const signedInChildProps = {
			email: user && user.email || 'email',
		};

		return loggedIn ? (
			<SignedInView {...signedInChildProps} />
		) : (
			<LogInView {...logInChildProps} />
		);
	}
}

// PropTypes ensure we pass required props of the correct type
LogInViewContainer.propTypes = {
	actions: PropTypes.shape({
		setToast: PropTypes.func.isRequired,
		login: PropTypes.func.isRequired,
		getUser: PropTypes.func.isRequired,
		getUserSettings: PropTypes.func.isRequired,
	}).isRequired,
};

export default LogInViewContainer;
