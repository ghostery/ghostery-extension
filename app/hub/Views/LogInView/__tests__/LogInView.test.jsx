/**
 * LogIn View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import LogInView from '../LogInView';

describe('app/hub/Views/LogIn component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('login view is rendered correctly', () => {
			const initialState = {
				email: 'test@example.com',
				password: 'examplePassword',
				emailError: false,
				passwordError: false,
				handleSubmit: () => {},
				handleInputChange: () => {},
			};

			const component = renderer.create(
				<MemoryRouter>
					<LogInView {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('login view is rendered correctly with empty strings and errors', () => {
			const initialState = {
				email: '',
				password: '',
				emailError: true,
				passwordError: true,
				handleSubmit: () => {},
				handleInputChange: () => {},
			};

			const component = renderer.create(
				<MemoryRouter>
					<LogInView {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const initialState = {
				email: 'test@example.com',
				password: 'examplePassword',
				emailError: false,
				passwordError: false,
				handleSubmit: jest.fn(),
				handleInputChange: () => {},
			};

			const component = shallow(<LogInView {...initialState} />);
			expect(component.find('.LogInView').length).toBe(1);
			expect(component.find('.LogInView__inputBox').length).toBe(2);
			expect(component.find('button[type="submit"]').length).toBe(1);

			expect(component.find('.LogInView__inputError').length).toBe(0);
			component.setProps({ emailError: true });
			expect(component.find('.LogInView__inputError').length).toBe(1);

			expect(initialState.handleSubmit.mock.calls.length).toBe(0);
			component.find('form').simulate('submit');
			expect(initialState.handleSubmit.mock.calls.length).toBe(1);
		});
	});
});
