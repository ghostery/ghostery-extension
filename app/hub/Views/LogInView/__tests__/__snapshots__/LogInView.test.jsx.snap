// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/LogIn component Snapshot tests with react-test-renderer login view is rendered correctly 1`] = `
<div
  className="LogInView"
>
  <div
    className="LogInView--addPaddingTop row align-center"
  >
    <div
      className="columns"
    >
      <div
        className="LogInView__header flex-container align-center-middle"
      >
        <img
          className="LogInView__headerImage"
          src="/app/images/hub/account/ghosty-account.svg"
        />
        <div
          className="LogInView__headerTitle text-center"
        >
          <h3>
            hub_login_header_title
          </h3>
        </div>
      </div>
    </div>
  </div>
  <div
    className="LogInView--addPaddingTop row align-center"
  >
    <div
      className="columns small-12 medium-10 large-6"
    >
      <form
        onSubmit={[Function]}
      >
        <label
          className="LogInView__inputLabel"
          htmlFor="login-email"
        >
          email_colon
        </label>
        <input
          autoComplete="off"
          className="LogInView__inputBox"
          id="login-email"
          name="email"
          onChange={[Function]}
          pattern=".{1,}"
          type="text"
          value="test@example.com"
        />
        <label
          className="LogInView__inputLabel"
          htmlFor="login-password"
        >
          password_colon
        </label>
        <input
          className="LogInView__inputBox"
          id="login-password"
          name="password"
          onChange={[Function]}
          pattern=".{1,}"
          type="password"
          value="examplePassword"
        />
        <div
          className="flex-container align-center"
        >
          <span
            className="LogInView__link text-center"
          >
            <a
              aria-current={null}
              href="/forgot-password"
              onClick={[Function]}
            >
              forgot_password
            </a>
          </span>
        </div>
        <div
          className="flex-container align-center"
        >
          <span
            className="LogInView__link text-center"
          >
            hub_login_link_dont_have_account
             
            <a
              aria-current={null}
              href="/create-account"
              onClick={[Function]}
            >
              hub_login_link_create_account
            </a>
          </span>
        </div>
        <div
          className="LogInView--addPaddingTop flex-container align-center"
        >
          <button
            className="LogInView__button button success"
            type="submit"
          >
            sign_in
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`app/hub/Views/LogIn component Snapshot tests with react-test-renderer login view is rendered correctly with empty strings and errors 1`] = `
<div
  className="LogInView"
>
  <div
    className="LogInView--addPaddingTop row align-center"
  >
    <div
      className="columns"
    >
      <div
        className="LogInView__header flex-container align-center-middle"
      >
        <img
          className="LogInView__headerImage"
          src="/app/images/hub/account/ghosty-account.svg"
        />
        <div
          className="LogInView__headerTitle text-center"
        >
          <h3>
            hub_login_header_title
          </h3>
        </div>
      </div>
    </div>
  </div>
  <div
    className="LogInView--addPaddingTop row align-center"
  >
    <div
      className="columns small-12 medium-10 large-6"
    >
      <form
        onSubmit={[Function]}
      >
        <label
          className="LogInView__inputLabel"
          htmlFor="login-email"
        >
          email_colon
        </label>
        <input
          autoComplete="off"
          className="LogInView__inputBox error"
          id="login-email"
          name="email"
          onChange={[Function]}
          pattern=".{1,}"
          type="text"
          value=""
        />
        <div
          className="LogInView__inputError"
        >
          please_enter_a_valid_email
        </div>
        <label
          className="LogInView__inputLabel"
          htmlFor="login-password"
        >
          password_colon
        </label>
        <input
          className="LogInView__inputBox error"
          id="login-password"
          name="password"
          onChange={[Function]}
          pattern=".{1,}"
          type="password"
          value=""
        />
        <div
          className="LogInView__inputError"
        >
          hub_login_label_password_invalid
        </div>
        <div
          className="flex-container align-center"
        >
          <span
            className="LogInView__link text-center"
          >
            <a
              aria-current={null}
              href="/forgot-password"
              onClick={[Function]}
            >
              forgot_password
            </a>
          </span>
        </div>
        <div
          className="flex-container align-center"
        >
          <span
            className="LogInView__link text-center"
          >
            hub_login_link_dont_have_account
             
            <a
              aria-current={null}
              href="/create-account"
              onClick={[Function]}
            >
              hub_login_link_create_account
            </a>
          </span>
        </div>
        <div
          className="LogInView--addPaddingTop flex-container align-center"
        >
          <button
            className="LogInView__button button success"
            type="submit"
          >
            sign_in
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
